Replace use of `stat` in imgmount

shermp opened the pull request:
# Description

Replace use of `stat` in `imgmount` with `path_exists` and `is_directory`, which fixes mounting large disk image files with Visual Studio builds.

## Related issues
Fixes #3464 

# Manual testing

If you don't already have one, create a large `img` file (tested with 8GB image).

Start Dosbox-Staging, and enter the following imgmount command:
```
imgmount 2 "<path-to-disk-image>" -fs none -t hdd -size <size-for-disk-image>
```
Dosbox should show the message `Drive number 2 mounted as <image-path>`.

Also check the non existing file case:
```
imgmount 2 "<path-to-nonexisting-file>" -fs none -t hdd -size 512,63,255,1023
```
Dosbox should show an error message.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [ ] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

shermp commented:
This whole bit of code could probably use a refactor at some point, but this at least fixes the linked issue.

weirddan455 commented on src/dos/program_imgmount.cpp:
```diff
@@ -342,7 +340,7 @@ void IMGMOUNT::Run(void)
-		if (S_ISDIR(test.st_mode)) {
+		if (is_directory(temp_line)) {
 			WriteOut(MSG_Get("PROGRAM_IMGMOUNT_MOUNT"));
 			return;
 		}
```
Looks like you fixed a bug here too.  Before, if that first `stat` call failed, the value of `struct stat test` is possibly undefined (we didn't zero clear it).

weirddan455 reviewed (approved):
Looks like a good change to me.  Yeah this whole function is kind of hard to read but it doesn't have to be refactored right now if you don't feel like it.

shermp commented:
Um... Oops, I think I got this wrong, and I'm not sure how this works. Give me a few minutes to fix my boo-boo's.

weirddan455 commented:
Oh you got the conditions inverted.  I missed that too.  Should be checking if the path doesn't exist.

shermp commented:
Ok, I've fixed my conditions. I've also tested the file not exist path (I updated the PR description).

@weirddan455 can you please test on Linux to make sure nothing has broken there?

weirddan455 commented:
Tested `imgmount` command on Linux and it looks good.

shermp commented:
> Tested `imgmount` command on Linux and it looks good.

Thanks. Merging.

shermp commented:
Actually, someone else needs to merge. I do not have permissions to do so.

weirddan455 commented:
You pushed again, restarting CI.  The button is grayed out for me too until that finishes.
